stdlib/public/TensorFlow/Ops.swift

```diff
@@ -1597,52 +1597,302 @@ public extension Tensor where Scalar : Numeric {
+      return r1.upperBound == r2.upperBound && s1 == s2
+    case (let .partialRangeThrough(r1, s1), let .partialRangeThrough(r2, s2)):
+      return r1.upperBound == r2.upperBound && s1 == s2
+    default: return false
+    }
+  }
+}
+
+public protocol TensorRangeExpression {
+  var tensorRange: TensorRange { get }
+}
+
+// TODO: Cannot extend non-nominal type 'UnboundedRange'.
+// extension UnboundedRange : TensorRangeExpression {
+//   public var tensorRange: TensorRange { return .ellipsis }
+// }
+
+extension Int32 : TensorRangeExpression {
+  public var tensorRange: TensorRange { return .index(self) }
+}
+
+extension Int : TensorRangeExpression {
+  public var tensorRange: TensorRange { return .index(Int32(self)) }
+}
+
+extension Range : TensorRangeExpression where Bound == Int {
+  public var tensorRange: TensorRange {
+    return .range(Int32(lowerBound)..<Int32(upperBound), stride: 1)
+  }
+}
+
+extension ClosedRange : TensorRangeExpression where Bound == Int {
+  public var tensorRange: TensorRange {
+    return .closedRange(
+      Int32(lowerBound)...Int32(upperBound), stride: 1)
+  }
+}
+
+extension PartialRangeFrom : TensorRangeExpression where Bound == Int {
+  public var tensorRange: TensorRange {
+    return .partialRangeFrom(Int32(lowerBound)..., stride: 1)
+  }
+}
+
+extension PartialRangeUpTo : TensorRangeExpression where Bound == Int {
+  public var tensorRange: TensorRange {
+    return .partialRangeUpTo(..<Int32(upperBound), stride: 1)
+  }
+}
+
+extension PartialRangeThrough : TensorRangeExpression where Bound == Int {
+  public var tensorRange: TensorRange {
+    return .partialRangeThrough(...Int32(upperBound), stride: 1)
+  }
+}
+
+infix operator ..: TensorRangePrecedence
+precedencegroup TensorRangePrecedence {
+  associativity: left
+  higherThan: CastingPrecedence
+  lowerThan: RangeFormationPrecedence
+}
+
+public extension Range where Bound == Int {
+  static func .. (range: Range, stride: Int) -> TensorRange {
+    return .range(
+      Int32(range.lowerBound)..<Int32(range.upperBound), stride: Int32(stride))
+  }
+}
+
+public extension ClosedRange where Bound == Int {
+  static func .. (range: ClosedRange, stride: Int) -> TensorRange {
+    return .closedRange(
+      Int32(range.lowerBound)...Int32(range.upperBound), stride: Int32(stride))
+  }
+}
+
+public extension PartialRangeFrom where Bound == Int {
+  static func .. (range: PartialRangeFrom, stride: Int) -> TensorRange {
+    return .partialRangeFrom(Int32(range.lowerBound)..., stride: Int32(stride))
+  }
+}
+
+public extension PartialRangeUpTo where Bound == Int {
+  static func .. (range: PartialRangeUpTo, stride: Int) -> TensorRange {
+    return .partialRangeUpTo(..<Int32(range.upperBound), stride: Int32(stride))
+  }
+}
+
+public extension PartialRangeThrough where Bound == Int {
+  static func .. (range: PartialRangeThrough, stride: Int) -> TensorRange {
+    return .partialRangeThrough(...Int32(range.upperBound), stride: Int32(stride))
```

Comment:
I think these can be changed to 
```swift
return .partialRangeThrough(range, stride: Int32(stride))
```
... once you merge the `Int32` -> `Int` change.